### PR TITLE
Feature/public snippets

### PR DESCRIPTION
**Pull request by dwoznicka**

### CODE REVIEW

File: client/js/components/app.jsx [hunk 1]
```diff
@@ -32,9 +32,10 @@ export default class App extends React.Component {
     this.setState({ currentUser: UserStore.getState().currentUser });
     setTimeout(() => {
       var currentPath = this.props.location.pathname;
+      var currentParamPath = this.props.routes[1].path;
 
-      if(!this.state.currentUser && currentPath != '/login' && currentPath != '/register') {
-        this.props.history.pushState(null, '/login');
+      if(!this.state.currentUser && currentPath != '/login' && currentPath != '/register' && currentPath != '/snippets' && currentParamPath != '/snippets/:id') {
```
Comment (Vakrim) [inline client/js/components/app.jsx]: are you sure that `currentParamPath != '/snippets/:id` part works correctly?

File: client/js/components/navbar.jsx [hunk 2]
```diff
@@ -23,7 +35,12 @@ class Navbar extends React.Component {
   }
 
   render() {
-    let iconElementRight = (<RaisedButton style={{marginTop: '6px'}} onClick={ () => this.refs.dialog.open()} label="Add new snippet" primary={true}/>);
+    let iconElementRight = '';
```
Comment (Vakrim) [inline client/js/components/navbar.jsx]: you don't need to assign value at declaration. `let iconElementRight;` works fine

File: client/js/components/navbar.jsx [hunk 3]
```diff
@@ -23,7 +35,12 @@ class Navbar extends React.Component {
...
+    if ( this.state.currentUser ){
+      iconElementRight = (<RaisedButton style={{marginTop: '6px'}} onClick={ () => this.refs.dialog.open()} label="Add new snippet" primary={true}/>);
```
Comment (Vakrim) [inline client/js/components/navbar.jsx]: you don't have to create anonymous function here. Just pass function. Try `onClick={ this.refs.dialog.open }`

**Comment by Vakrim** [comment]
:+1:

File: client/js/components/comments/comment-box.jsx [hunk 4]
```diff
@@ -4,16 +4,18 @@ import SnippetActions from '../../actions/snippet-actions';
 import CommentForm from './comment-form';
 import CommentList from './comment-list';
 import SnippetStore from '../../stores/snippet-store';
+import UserStore from '../../stores/user-store';
 
 export default class CommentBox extends React.Component {
   constructor(props) {
     super(props);
-    this.state = { comments: props.comments, content: '', createError: ''};
+    this.state = { comments: props.comments, content: '', createError: '', currentUser: UserStore.getState().currentUser };
```
Comment (pokorson) [inline client/js/components/comments/comment-box.jsx]: po co `currentUser` w state?

File: client/js/components/navbar.jsx [hunk 5]
```diff
@@ -11,6 +12,17 @@ class Navbar extends React.Component {
     super();
     this.handleClick = this.handleClick.bind(this);
     this.onLeftNavChange = this.onLeftNavChange.bind(this);
+    this._onChange = this._onChange.bind(this);
+    this.state = { currentUser: UserStore.getState().currentUser };
```
Comment (pokorson) [inline client/js/components/navbar.jsx]: po co `currentUser` w state?

File: client/js/components/snippets/snippets-list.jsx [hunk 6]
```diff
@@ -2,20 +2,30 @@ import React from 'react';
...
 import Snippet from './snippet';
 import CommentBox from '../comments/comment-box';
+import UserStore from '../../stores/user-store';
 
 class SnippetListItem extends React.Component{
   constructor(props) {
     super(props);
   }
+
+  getCurrentUser() {
```
Comment (pokorson) [inline client/js/components/snippets/snippets-list.jsx]: raz dodajesz user do `state` a raz wyciągasz sobie metodą? :)

File: server/routes/dashboard.js [hunk 7]
```diff
@@ -2,50 +2,46 @@ var express = require('express');
...
-      return models.Snippet.scope(['fromCurrentMonth', 'withVersions', 'withAuthor', { method: ['withRatings', req.user.get('id')] }])
-        .findAll({order: [ ['avg', 'DESC'], ['createdAt', 'DESC'] ], limit: 5 });
-    }).then(function (bests) {
-      bestSnippets = bests.map( function (s) {
-        return s.toJson();
-      });
-      res.status(200).send({ count: count, comments: mappedComments, snippets: mappedSnippets, bestSnippets: bestSnippets });
-    });
+  } else {
+    res.status(200).send({ count: 0, comments: [], snippets: [], bestSnippets: [] });
```
Comment (pokorson) [inline server/routes/dashboard.js]: nie powinniśmy wysyłać jakiegoś `401` na niezalogowany dashboard?

**Pull request merged**